### PR TITLE
Implement pcap logging for netty HTTP client

### PR DESCRIPTION
**Pull request by yawkat**
The server has supported this for a long time. This PR adds the same feature for client logging. The code is very similar, and the test is basically copied.

### CODE REVIEW

**Review by graemerocher — APPROVED** [review]

**Review by sdelamo — CHANGES_REQUESTED** [review]
Please, mention in https://docs.micronaut.io/4.6.5/guide/#jdkHttpClient that JDK Client implementation does it does not support pcap logging

**Comment by yawkat** [comment]
@dstepanov PTAL

File: http-client-core/src/main/java/io/micronaut/http/client/HttpClientConfiguration.java [hunk 1]
```diff
@@ -872,6 +875,28 @@ public HttpClientConfiguration.Http2ClientConfiguration getHttp2Configuration()
...
     }
 
+    /**
+     * The path pattern to use for logging outgoing connections to pcap. This is an unsupported option: Behavior may
+     * change, or it may disappear entirely, without notice! Only implemented for netty.
+     *
+     * @return The path pattern, or {@code null} if logging is disabled.
+     */
+    @Internal
+    public String getPcapLoggingPathPattern() {
```
Comment (dstepanov) [inline http-client-core/src/main/java/io/micronaut/http/client/HttpClientConfiguration.java:885]: What's the point of having the accessors internal? People will not see the property configuration as an internal

**Review by dstepanov — APPROVED** [review]

File: http-client-core/src/main/java/io/micronaut/http/client/HttpClientConfiguration.java (hunk 1 above)
Comment (yawkat) [reply http-client-core/src/main/java/io/micronaut/http/client/HttpClientConfiguration.java:885]: the comment is for config users, the annotation is for programmatic users.

**Pull request merged**